src/Bicep.LangServer/Completions/ParamsCompletionContext.cs

```diff
@@ -53,21 +53,28 @@ public static ParamsCompletionContext Create(ParamsCompilationContext paramsComp
 
 
             //TODO: Add top level declaration completions
-            var kind = ConvertFlag(IsParamAssignmentContext(matchingNodes, offset), ParamsCompletionContextKind.ParamAssignment);
+            var kind = ConvertFlag(IsParamAssignmentContext(matchingNodes, offset), ParamsCompletionContextKind.ParamAssignment) |
+                        ConvertFlag(IsUsingDeclarationContext(matchingNodes, offset), ParamsCompletionContextKind.UsingDeclaration);
 
             return new(kind, replacementRange);
         }
 
+        private static bool IsUsingDeclarationContext(List<SyntaxBase> matchingNodes, int offset)
+        {
+            if(matchingNodes.Count >=1)
+            {
+                SyntaxBase lastNode = matchingNodes[^1];
+                return lastNode is UsingDeclarationSyntax usingDeclarationSyntax && usingDeclarationSyntax.Path is SkippedTriviaSyntax;
+            }
+            return false;
+        }
+
         private static bool IsParamAssignmentContext(List<SyntaxBase> matchingNodes, int offset)
         {
             if(matchingNodes.Count >=1)
             {
                 SyntaxBase lastNode = matchingNodes[^1];
-                if (lastNode is ParameterAssignmentSyntax assignmentSyntax )
-                {
-                    //TODO: do completions with partially written identifier names
-                    return assignmentSyntax.Name.IdentifierName == LanguageConstants.MissingName;
-                }
+                return lastNode is ParameterAssignmentSyntax assignmentSyntax && assignmentSyntax.Name.IdentifierName == LanguageConstants.MissingName;
```

Comment:
I'd recommend basing this implementation on https://github.com/Azure/bicep/blob/a2b797095ae00aac7ae87cf48fce791d477700f3/src/Bicep.LangServer/Completions/BicepCompletionContext.cs#L635-L643

Syntactically it's almost identical, and there are a few edge cases that this implementation handles.

Reply:
These are the scenarios I came up with 

https://github.com/Azure/bicep/blob/9d0a2af42b5ff332b2ca15fbc53911f24526d5bd/src/Bicep.LangServer/Completions/ParamsCompletionContext.cs#L76-L82